authors/apps/profiles/views.py

```diff
@@ -0,0 +1,57 @@
+from rest_framework.generics import (
+    ListAPIView,
+    RetrieveUpdateAPIView)
+from rest_framework.permissions import (
+    IsAuthenticatedOrReadOnly,
+    AllowAny
+)
+from .renderers import ProfileJSONRenderer
+from .permissions import IsOwnerOrReadOnly
+from .serializers import ProfileSerializer
+from .models import Profile
+from .exceptions import ProfileDoesNotExist
+from rest_framework import status
+from rest_framework.response import Response
+
+
+class ListProfileView(ListAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    queryset = Profile.objects.all()
+    serializer_class = ProfileSerializer
+    ordering_fields = ('timestamp')
+
+
+class RetrieveUpdateProfileView(RetrieveUpdateAPIView):
+    """
+    This method allows a user to view a profile
+    and only its owner to edit it
+    """
+    queryset = Profile.objects.all()
+    serializer_class = ProfileSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,
+                          IsOwnerOrReadOnly, AllowAny,)
+    renderer_classes = (ProfileJSONRenderer,)
+
+    def check_for_object(self, username):
```

Comment:
@NLSanyu Instead of you having check_for_object just override the get_object method which is already implemented by the framework. Reach out to @walimike but I do like the logic.